.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,72 @@
+name: Benchmarks
+
+on:
+  pull_request:
+    branches: [ '**' ]
+    types: [opened, synchronize, reopened, labeled]
+
+env:
+  PROMSCALE_LOGGING: false
+  PROMSCALE_BENCHMARK: true
+  golang-version: 1.18.4
+
+jobs:
+  benchmark:
+    if: contains( github.event.pull_request.labels.*.name, 'action:benchmarks')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v3.2.1
+        with:
+          go-version: ${{ env.golang-version }}
+        id: go
+
+      - name: Benchmark current PR
+        run: go test -bench=. -run=^$ -benchmem -cpu=1 ./... | tee current.txt
```

Comment:
The only concern I have is that with only 1 run, we may not get concrete enough results to be able to determine if there was a difference or not.

Reply:
Do you mean, with `-cpu=1` or with `-benchtime=<some fixed iterations>`?

Reply:
I meant the `-count` parameter. Implicitly we're setting `-count=1`. I'm not sure that I understand it well yet, but I've had situations where count was low, and benchstat would basically say that it didn't have enough data to determine if the outcome was statistically significant or not.